pkg/didcomm/common/service/messenger.go

```diff
@@ -28,12 +28,25 @@ type Messenger interface {
 	// ReplyTo replies to the message by given msgID.
 	// Keeps threadID in the *decorator.Thread.
 	// Using this function means that communication will be on the same thread.
-	ReplyTo(msgID string, msg DIDCommMsgMap)
+	ReplyTo(msgID string, msg DIDCommMsgMap) error
 
 	// Send sends the message by starting a new thread.
-	Send(msg DIDCommMsgMap, myDID, theirDID string)
+	Send(msg DIDCommMsgMap, myDID, theirDID string) error
 
 	// ReplyToNested sends the message by starting a new thread.
 	// Keeps parent threadID in the *decorator.Thread
-	ReplyToNested(threadID string, msg DIDCommMsgMap)
+	ReplyToNested(threadID string, msg DIDCommMsgMap, myDID, theirDID string) error
+}
+
+// MessengerHandler includes Messenger interface and Handle function to handle inbound messages
+type MessengerHandler interface {
+	Messenger
+	// HandleInbound handles all inbound messages
+	HandleInbound(msg DIDCommMsgMap, myDID, theirDID string, opts ...interface{}) error
+}
+
+// MessengerOptions allows modifying Messenger behavior
+// NOTE: if service implements it, it will work automatically
+type MessengerOptions interface {
```

Comment:
I'm not sure what this interface does?

Reply:
if for some reason you do want to turn messenger logic off or change the behavior of how it works you can do this by implementing this interface in your service. Messenger will check whether the service implements it and will act according to that. Possible options will be implemented in the messenger. For now, we do not have any options. Implementing this interface in the service is optional.  It intends to provide a possibility to customize the messenger behavior in case you are advanced user :)

Reply:
can we defer this until needed?

Reply:
sure, I can remove this

Reply:
agree - remove until needed.

Reply:
Done!